geode-core/src/main/java/org/apache/geode/cache/execute/FunctionContext.java

```diff
@@ -77,4 +80,22 @@
   public boolean isPossibleDuplicate();
 
   public Cache getCache();
+
+  /**
+   * a convenience method to get the name of the member this function executes on. call this
+   * function once in your function execution to avoid performance issues.
+   *
+   * @return member name or id if name is blank
+   */
+  public default String getMemberName() {
```

Comment:
I don't think that this is the right place for this. It feels like it's just a utility method and would never actually be overridden by implementations of this interface.

Reply:
May feel like a utility method, but I think it's appropriate. This same bit of code is exists  in other *Function classes so it makes sense to put the implementation here.

Reply:
I can see how it would help with mocking/testing, but it still smells a bit to me. :)